src/components/DateRangePickerInput.jsx

```diff
@@ -172,16 +192,9 @@ export default class DateRangePickerInput extends React.Component {
           'DateRangePickerInput--rtl': isRTL,
         })}
       >
-        {(showDefaultInputIcon || customInputIcon !== null) && (
-          <button
-            type="button"
-            className="DateRangePickerInput__calendar-icon"
-            disabled={disabled}
-            aria-label={phrases.focusStartDate}
-            onClick={onArrowDown}
-          >
-            {inputIcon}
-          </button>
+
+        {(inputIconPosition === 'before' && (showDefaultInputIcon || customInputIcon !== null)) && (
```

Comment:
We could probably pull `(showDefaultInputIcon || customInputIcon !== null) && this.renderInputIcon(inputIcon)` into a variable above the `return` and then just have `inputIconPosition === 'before' && inputIcon`